addon/components/select-component.js

```diff
@@ -482,17 +488,21 @@ export default Ember.Component.extend(
 
   // The option that is currently highlighted.
   highlighted: Ember.computed(function(key, value) {
-    var content, index;
-    content = this.get('selectableOptions') || Ember.A();
-    value = value || Ember.A();
+    let content = this.get('selectableOptions') || Ember.A();
+
     if (arguments.length === 1) {
-      index = this.get('highlightedIndex');
-      value = content.objectAt(index);
-    } else {
-      index = content.indexOf(value);
-      this.setHighlightedIndex(index, this.get('shouldEnsureVisible'));
-    }
-    return value;
+      // Getter
+      return content.objectAt(this.get('highlightedIndex'));
+    }
+    
+    // Setter
+    let index = value ? content.indexOf(value) : -1;
+    this.setHighlightedIndex(
+      index,
+      this.get('shouldEnsureVisible')
+    );
+    return value || [];
```

Comment:
I see that the `|| []` isn't a change introduced by this PR, but is there any rationale for why `highlighted` should be an empty array?